stages/org.osbuild.oscap.remediation

```diff
@@ -128,15 +129,82 @@ def main(tree, options):
     if tailoring is not None:
         cmd.extend(["--tailoring-file", tailoring])
 
+    if arf_results is not None:
+        # run with command in chroot so full path ok
+        cmd.extend(["--results-arf", f"{data_dir}/{arf_results}"])
+
+    cmd.extend(["--results", f"{data_dir}/{xccdf_results}"])
+
     cmd.append(datastream)
 
-    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False)
+    print("OpenSCAP (evaluate): ", repr(cmd))
+    # we need to inform the scanner that the system is offline (OSCAP_PROBE_ROOT="")
+    # and provide environment variables in OSCAP_CONTAINER_VARS as the probe
+    # does not have access to /proc
+    res = subprocess.run(cmd, encoding="utf8", stdout=sys.stderr, check=False,
+                         env=dict(os.environ,
+                             OSCAP_PROBE_ROOT="",
+                             OSCAP_CONTAINER_VARS="container=bwrap-osbuild"))
 
     # oscap return values are:
     # 0 → success
-    # 2 → -- no error, but some checks/remediation failed
+    # 2 → no error, but some checks/remediation failed
+    if res.returncode not in (0, 2):
+        raise RuntimeError("oscap content evaluation failed")
+
+    # run openscap in chroot on the image tree (generate remediation script)
+    cmd = [
+        "/usr/sbin/chroot", tree,
+        "/usr/bin/oscap",
+        "xccdf", "generate"
```

Comment:
I am wondering if we should maybe at this point split this into 3 stages then?
- scan
- generate
- remediate

osbuild-composer would take care of passing through these parameters and we could handle the `data-dir` and results from there.

@supakeen & @thozza, what are your thoughts? I'm assuming this might be a backwards compatibility issue?

Reply:
It might not be necessary in the future. We just can't do this change in the scanner right away. Also, it'd require change in OpenSCAP to generate new oscap entry(ies) for the blueprint.

Reply:
@kingsleyzissou splitting the stage into three pieces is not a problem from backwards compatibility PoV, as long as the original stage keeps working as it did before. This would probably mean introducing three new stages.

Nevertheless, my opinion on whether to split the stage or not is that it would make sense only if each of those new stages would be useful on its own. If one would need to always use all three of those stages in combination to get some useful result, then it makes more sense to keep the implementation within a single stage.

Reply:
Yeah, very unlikely that they would be used in isolation from each other, so we can keep them as is then :)